queries.sql

```diff
@@ -154,3 +210,126 @@ FROM
     ON owners.id = animals.owner_id 
   GROUP BY owners.full_name 
   ORDER BY COUNT(animals.name) DESC LIMIT 1) number ;
+
+select * from visits where vets_id=1 order by date_of_visit desc limit 1;
+
+/*Who was the last animal seen by William Tatcher?*/
+SELECT
+  animals.name,
+  date_of_visit
+FROM
+  animals
+INNER JOIN visits
+  ON animals.id = animal_id
+WHERE
+  vets_id = (SELECT id FROM vets WHERE name LIKE 'William Tatcher')
+ORDER BY date_of_visit DESC LIMIT 1;
+
+/*How many different animals did Stephanie Mendez see?*/
+SELECT
+  count(animal_id) as number_of_animals
+FROM
+  visits
+WHERE
+  vets_id = (SELECT id FROM vets WHERE name LIKE 'Stephanie Mendez');
+
+-- List all vets and their specialties, including vets with no specialties.
+SELECT
+  vets.name as vet_name,
+  species.name as specilization
+FROM
+  vets
+LEFT JOIN specializations
+  ON vets.id = vets_id
+LEFT JOIN species
+  ON specializations.species_id = species.id;
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT
+  animals.name as animal_name,
+  date_of_visit
+FROM
+  visits
+JOIN animals
+  ON animal_id = animals.id
+WHERE 
+  (date_of_visit BETWEEN '2020-04-01' AND '2020-08-30') AND
+  visits.vets_id = (SELECT id FROM vets WHERE vets.name = 'Stephanie Mendez')
+ORDER BY date_of_visit ASC;
+
+-- What animal has the most visits to vets?
+SELECT
+  animals.name as animal_name,
+  MAX(animal_id) as number_of_visits
+FROM
+  visits
+JOIN animals
+  ON animal_id = animals.id
+GROUP BY animals.name
+ORDER BY number_of_visits DESC LIMIT 1;
+
+-- Who was Maisy Smith's first visit?
+SELECT
+  animals.name as animal_name
+FROM
+  visits
+JOIN animals
+  ON animal_id = animals.id
+WHERE vets_id = (SELECT id FROM vets WHERE name = 'Maisy Smith')
+ORDER BY date_of_visit LIMIT 1;
+
+
+-- Details for most recent visit: animal information, vet information, and date of visit.
+SELECT
+  animals.name as animal_name,
+  animals.date_of_birth,
+  animals.escape_attempts,
+  animals.neutered as is_neutered,
+  animals.weight_kg,
+  species.name as species,
+  owners.full_name as owner_full_name,
+  vets.name as vet_name,
+  vets.age as vet_age,
+  vets.date_of_graduation as vet_date_of_graduation,
```

Comment:
- [ ] (Optional) There is a shorthand syntax for this. It is `animals.*, vets.*,`

Reply:
Thanks @Gambit142 for the comment, I did it the way I did so I can change the names of the columns to give a better view later, since the word name gets repeated several times, and I also didn't use * because I didn't want to show the id of the rows.

Reply:
@mirouhml, no problem. I understand. I love the work you did with the insertion of data. 👍

Reply:
Thanks a lot, I appreciate it!